core/server/master/src/main/java/alluxio/master/meta/ServerConfigurationChecker.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.meta;
+
+import alluxio.wire.ConfigProperty;
+
+import java.util.List;
+
+/**
+ * This class is responsible for checking server-side configuration.
+ */
+public final class ServerConfigurationChecker {
+  private ConfigRecorder mMasterConfigRecorder;
+  private ConfigRecorder mWorkerConfigRecorder;
```

Comment:
instead of storing two `ConfigRecorders` in this class and adding an `isMaster` parameter to every method, could we instead have two instances of `ServerConfigurationChecker`, one that we use for masters and one that we use for workers? Then `ServerConfigurationChecker` would only hold one `ConfigRecorder` and we wouldn't need any of the `isMaster` stuff.